### PR TITLE
Hyperzine rebalance mk2

### PR DESCRIPTION
**Pull request by jroinc**
## About The Pull Request

Re-balances hyperzine so it's actually usable in combat.

## Why It's Good For The Game

Hyperzine wasn't usable because of a 20-30 second stun AND massive damage.

## Changelog
:cl:
balance: Lowers hyperzine damage when it runs out to 10 pts oxy and 12 pts halloss/unit.
Slightly lowers max attainable speed to ultrazine levels.
Allows dexalin+ to be used in hyperzine blends.
add: Adds 2 different autoinjectors-
 -Hyperzine injectors, available from medic prep for 15 points offer a safe speedboost with some hunger as the only downside.
-Expired Hyperzine injectors, available as contraband in marinemeds offer a speedboost for the cost of 5 brain and 10 toxin damage. Also hunger.
/:cl:

Tested all autoinjectors and spawning locations.
Tested hyperzine effects and non-purging of dex+.

### CODE REVIEW

File: code/game/objects/items/reagent_containers/autoinjectors.dm [hunk 1]
```diff
@@ -109,3 +109,25 @@
 	volume = 3
 	icon_state = "autoinjector-8" //TEMP
 	list_reagents = list(/datum/reagent/medicine/hypervene = 3)
+
+/obj/item/reagent_container/hypospray/autoinjector/hyperzine_expired
```
Comment (spookydonut) [inline code/game/objects/items/reagent_containers/autoinjectors.dm]:
```suggestion
/obj/item/reagent_container/hypospray/autoinjector/hyperzine/expired
```

File: code/game/objects/items/reagent_containers/autoinjectors.dm [hunk 2]
```diff
@@ -109,3 +109,25 @@
...
+	name = "expired hyperzine autoinjector"
+	desc = "An auto-injector said to be loaded with a safe-to-use hyperzine mix, 3 months past it's expiration date."
+	amount_per_transfer_from_this = 25
```
Comment (spookydonut) [inline code/game/objects/items/reagent_containers/autoinjectors.dm]:
```suggestion
```

File: code/game/objects/items/reagent_containers/autoinjectors.dm [hunk 3]
```diff
@@ -109,3 +109,25 @@
...
+	volume = 25
```
Comment (spookydonut) [inline code/game/objects/items/reagent_containers/autoinjectors.dm]:
```suggestion
```

File: code/game/objects/items/reagent_containers/autoinjectors.dm [hunk 4]
```diff
@@ -109,3 +109,25 @@
...
+	icon_state = "autoinjector-1" //TEMP, god willing
```
Comment (spookydonut) [inline code/game/objects/items/reagent_containers/autoinjectors.dm]:
```suggestion
```

File: code/game/objects/machinery/vending/marine_vending.dm [hunk 5]
```diff
@@ -391,7 +392,8 @@
 					/obj/item/healthanalyzer = 3,
 					/obj/item/bodybag/cryobag = 2)
 
-	contraband = list(/obj/item/reagent_container/hypospray/autoinjector/sleeptoxin =3)
+	contraband = list(/obj/item/reagent_container/hypospray/autoinjector/sleeptoxin =3,
+					/obj/item/reagent_container/hypospray/autoinjector/hyperzine_expired =3)
```
Comment (spookydonut) [inline code/game/objects/machinery/vending/marine_vending.dm]:
```suggestion
					/obj/item/reagent_container/hypospray/autoinjector/hyperzine/expired =3)
```

**Comment by jroinc** [comment]
I took a look at how good it was, and decided it needed a nerf, even before playtesting. Knockdown amount doubled, halloss set to 15/u instead of 12/u, and movedelay reduction significantly reduced. You wanna hit the cap and be speed, you're gonna have to OD and actually take some damage now. That said, I'm planning more chem alterations which will interact with this in the future.

**Comment by jroinc** [comment]
Playtest was pretty good. Provides a moderate speed boost with around a 5 second stun, but the person can be shaken up in 2 shakes. Double-dosing provides a sanic speed boost, but torpedo's your heart.

**Comment by jroinc** [comment]
I commented out the code that was triggering it, but left it in.

It's currently unneeded code (due to anti-powergame reaction) that I want to reuse at a later time.

File: code/modules/reagents/chemistry_reagents/medical.dm [hunk 6]
```diff
@@ -610,16 +610,21 @@ datum/reagent/medicine/synaptizine/overdose_crit_process(mob/living/L, metabolis
...
+				L.knock_down(amount * 0.10)
 				to_chat(L, "<span class='danger'>You collapse as a sudden wave of fatigue washes over you.</span>")
 			if(50 to INFINITY)
 				L.knock_out(amount * 0.1)
 				to_chat(L, "<span class='danger'>Your world convulses as a wave of extreme fatigue washes over you!</span>") //when hyperzine is removed from the body, there's a backlash as it struggles to transition and operate without the drug
 
 	return ..()
 
+
+//datum/reagent/medicine/hyperzine/on_mob_add(mob/living/L, metabolism)
```
Comment (spookydonut) [inline code/modules/reagents/chemistry_reagents/medical.dm]: delete this stuff dont comment it out

Comment (jroinc) [reply code/modules/reagents/chemistry_reagents/medical.dm]: deleted

**Pull request merged**